CyvoreOS/phone_numbers_extension.py

```diff
@@ -0,0 +1,270 @@
+        len(matches_us),
+        len(matches_israel),
+        len(matches_europe),
+    )
+    for match in matches_us + matches_israel + matches_europe:
+        rgx_possible.add((match.strip()).lstrip("+"))
+    return list(rgx_possible)
+
+
+def normalize_phone_number(phone: str) -> str:
+    """
+    Normalizes a phone number by removing non-numeric symbols (except '+')
+    and ensuring it starts with a '+' if not already present.
+            (phonenumbers lib checks numbers better with + at start)
+    """
+    # Remove all characters except digits and '+'
+    normalized = re.sub(r"[^\d\+]", "", phone)
+    logging.debug("After removing non-numeric symbols: %s", normalized)
+
+    # Ensure the phone number starts with '+'
+    if not normalized.startswith("+"):
+        normalized = "+" + normalized
+        logging.debug("Added '+' to the start of the number: %s", normalized)
+
+    logging.info("Final normalized phone number: %s", normalized)
+    return normalized
+
+
+def normalize_phone_numbers(phone_list: PhoneNumbersList) -> PhoneNumbersList:
+    """
+    Normalizes a list of phone numbers, ensuring each number starts with '+'
+    and contains only valid characters for phone numbers.
+    """
+    logging.info(
+        "Starting normalization of phone numbers. Total numbers: %d", len(phone_list)
+    )
+    normal = set()
+    for phone in phone_list:
+        logging.debug("Normalizing number: %s", phone)
+        normalized_phone = normalize_phone_number(phone)
+        normal.add(normalized_phone)
+        logging.debug("Normalized number added to the set: %s", normalized_phone)
+
+    normalized_list = list(normal)
+    logging.info(
+        "Normalization complete. Total unique normalized numbers: %d",
+        len(normalized_list),
+    )
+    return normalized_list
+
+
+# ------------------------------------------------------------------------------------------------------
+def validate_number(num: phonenumbers.PhoneNumber) -> str:
+    """
+    Validates a PhoneNumber object from the phonenumbers library.
+    Checks if the number is possible and valid. Returns the formatted number in E.164 format if valid.
+    """
+    # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    try:
+        logging.debug(
+            "Validating phone number: %s",
+            phonenumbers.format_number(num, phonenumbers.PhoneNumberFormat.E164),
+        )
+    except Exception as e:
+        logging.warning("Error formatting number for logging: %s", e)
+    # ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    if phonenumbers.is_possible_number(num):
+        logging.debug("The number is a possible number.")
+        if phonenumbers.is_valid_number(num):
+            logging.info("The number is valid.")
+            return phonenumbers.format_number(num, phonenumbers.PhoneNumberFormat.E164)
+        else:
+            logging.warning("The number is possible but not valid.")
+            return "Error - after is_valid_number"
```

Comment:
Don't return error string. use python Exceptions and raise error, and have try except when calling this function.